src/ProjectOrigin.WalletSystem.Server/Extensions/IEnumerableExtensions.cs

```diff
@@ -1,9 +1,47 @@
+using System;
 using System.Collections.Generic;
+using System.Globalization;
 using System.Linq;
+using ProjectOrigin.WalletSystem.Server.Models;
 
 namespace ProjectOrigin.WalletSystem.Server.Extensions;
 
 public static class IEnumerableExtensions
 {
     public static bool IsEmpty<TSource>(this IEnumerable<TSource> source) => !source.Any();
+
+    public static IEnumerable<IGrouping<string, TSource>> GroupByTime<TSource>(
```

Comment:
This is an overall comment about "holes". As far as I can see "holes" is not handled and will just be missing. So if it is a Day aggregation for a week and there is only data for Monday and Friday, then there will be no information about the missing days (Tue, Wed, Thur, Sat and Sun). 

We need to check with our client (EO frontend/Alexander) if it is an issue or not.

Reply:
well, this would not be something that could easily be done, because, is it a hole if you only bought production for hours? I would say if you have this sort of usecase, then one should use api's to pull raw data.

Reply:
I have talked with Alexander and he said the following about if he could handle "holes":

`Yes det er ikke noget problem sålænge jeg kan udlede hvilke dage det er jeg får data på `

So this should not be a problem